lib/debugger/debugger.dart

```diff
@@ -0,0 +1,1309 @@
+      resumeButton.disabled = !isPaused;
+    });
+
+    PButton stepOver, stepIn, stepOut;
+
+    final BreakOnExceptionControl breakOnExceptionControl =
+        new BreakOnExceptionControl();
+    breakOnExceptionControl.onPauseModeChanged.listen((String mode) {
+      debuggerState.setExceptionPauseMode(mode);
+    });
+    debuggerState.onExceptionPauseModeChanged.listen((String mode) {
+      breakOnExceptionControl.exceptionPauseMode = mode;
+    });
+
+    consoleArea = new ConsoleArea();
+
+    mainDiv.add(<CoreElement>[
+      div(c: 'section')
+        ..flex()
+        ..layoutHorizontal()
+        ..add(<CoreElement>[
+          div(c: 'debugger-menu')
+            ..layoutVertical()
+            ..add(<CoreElement>[
+              _buildMenuNav(),
+            ]),
+          div()
+            ..element.style.overflowX = 'hidden'
+            ..layoutVertical()
+            ..flex()
+            ..add(<CoreElement>[
+              div(c: 'section')
+                ..layoutHorizontal()
+                ..add(<CoreElement>[
+                  resumeButton,
+                  div(c: 'btn-group margin-left')
+                    ..add(<CoreElement>[
+                      stepIn = new PButton(null)
+                        ..add(<CoreElement>[
+                          span(c: 'octicon octicon-chevron-down'),
+                          span(text: 'Step in'),
+                        ])
+                        ..small(),
+                      stepOver = new PButton(null)
+                        ..add(<CoreElement>[
+                          span(c: 'octicon octicon-chevron-right'),
+                          span(text: 'Step over'),
+                        ])
+                        ..small(),
+                      stepOut = new PButton(null)
+                        ..add(<CoreElement>[
+                          span(c: 'octicon octicon-chevron-up'),
+                          span(text: 'Step out'),
+                        ])
+                        ..small(),
+                    ]),
+                  div()..flex(),
+                  breakOnExceptionControl,
+                ]),
+              sourceArea = div(c: 'section table-border')
+                ..flex()
+                ..layoutVertical()
+                ..add(<CoreElement>[
+                  _sourcePathDiv = div(c: 'source-head'),
+                ]),
+              div(c: 'section table-border secondary-area')
+                ..layoutVertical()
+                ..add(consoleArea.element),
+            ]),
+        ]),
+    ]);
+
+    _sourcePathDiv.setInnerHtml('&nbsp;');
+
+    debuggerState.onSupportsStepping.listen((bool value) {
+      stepOver.enabled = value;
+      stepIn.enabled = value;
+      stepOut.enabled = value;
+    });
+
+    stepOver.click(() => debuggerState.stepOver());
+    stepIn.click(() => debuggerState.stepIn());
+    stepOut.click(() => debuggerState.stepOut());
+
+    final Map<String, dynamic> options = <String, dynamic>{
```

Comment:
nit: consider omitting they types on the RHS unless they are adding value. we aren't forcing the flutter types everywhere style for this code anymore

Reply:
👍 will do; I started this PR before the style change

Reply:
FWIW I think it's nicer to omit types on the left. If the generic arguments are needed (they could be skipped altogether here) put them on the literal only.